lib/manager/npm/extract/index.ts

```diff
@@ -54,7 +62,7 @@ export async function extractPackageFile(
   );
   const packageJsonVersion = packageJson.version;
   let yarnWorkspacesPackages;
-  if (packageJson.workspaces && packageJson.workspaces.packages) {
+  if (!is.array(packageJson.workspaces) && packageJson.workspaces?.packages) {
```

Comment:
Why is workspaces code getting touched here as well - is it related to resolutions?

Reply:
It because of the type change of `NpmPackage`:
https://github.com/renovatebot/renovate/pull/5636/files#diff-a542e31527fa5987e653a2bd6c6208a2L5

Reply:
I can move that to a separate pr, but then this pr maybe would depend it

Reply:
Done. will rebase / merge when #5645 is merged